### PR TITLE
[ASDisplayNode] Majorly Improve Descriptions

### PR DESCRIPTION
**Pull request by Adlai-Holler**
I went to debug some issues today and got annoyed with how off-the-mark our object descriptions are.

### `po` ASTextNode
**Before:** `<ASTextNode 0x7ff7dc841a00 [_ASDisplayView : 0x7ff7dad37620]>`

Very little info, nonstandard format.

**After:** `<ASTextNode: 0x7fcdac01ec00; text = "Lick tail flick jump on the table stretching purr …"; frameInWindow = (106 350.5; 198 210); frame = (106 16; 198 210); cellNode = <KittenNode: 0x7fcdaae3f8c0>; interfaceState = { MeasureLayout | Preload | Display | Visible }; view = <_ASDisplayView: 0x7fcdac83fe80>>`

Way more information, including the frame onscreen, text and the nearest ASCellNode ancestor.

--- 

### `po` ASCellNode
**Before:** `<KittenNode 0x7ff7dac12350 name = Node for kitten #144 [_ASDisplayView : 0x7ff7dad1c0a0]>`

Very little info, nonstandard format.

**After:** `<KittenNode: 0x7fcdaae3f8c0; name = "Node for kitten #144"; frameInWindow = (0 334.5; 320 242); frame = (0 0; 320 242); interfaceState = { MeasureLayout | Preload | Display | Visible }; view = <_ASDisplayView: 0x7fcdac83f7e0>; indexPath = (0, 3); tableView = <ASTableView: 0x7fcdad01b000>>`

Way more information, including index path and owning ASTableView/ASCollectionView/ASTableNode/ASCollectionNode.

---

### `po` _ASDisplayView

**Before:** `<<ASTextNode: 0x7ff7dc841a00; text = "Lick quis nunc toss the mousie vel, tortor pellent…"; truncation string = "…"; frame = {{106, 16}, {198, 384.5}}; renderer = 0x7ff7dc399520>, view = <_ASDisplayView: 0x7ff7dad37620; frame = (106 16; 198 384.5); userInteractionEnabled = NO; layer = <<ASTextNode: 0x7ff7dc841a00; text = "Lick quis nunc toss the mousie vel, tortor pellent…"; truncation string = "…"; frame = {{106, 16}, {198, 384.5}}; renderer = 0x7ff7dc399520>, layer = <_ASDisplayLayer: 0x7ff7dad172d0>>>>`

Misleading. Looks like the description for a node. Way too many angle brackets, duplicated info, stupid irrelevant info, etc.

**After:** `<ASTextNode-View: 0x7fcdac83fe80; node = <ASTextNode: 0x7fcdac01ec00; text = "Lick tail flick jump on the table stretching purr …">; frame = (106 16; 198 210); userInteractionEnabled = NO>`

Immaculate.

---

Ready for review @levi @appleguy @maicki !

### CODE REVIEW

**Comment by Adlai-Holler** [comment]
I know there's some indentation troubs in here. I'll fix those before we land

File: AsyncDisplayKit/Private/_ASHierarchyChangeSet.mm [hunk 1]
```diff
@@ -454,7 +454,7 @@ - (void)_validateUpdate
     NSInteger insertedItemCount = originalInsertedItems.count;
     NSInteger deletedItemCount = originalDeletedItems.count;
     if (newItemCount != oldItemCount + insertedItemCount - deletedItemCount) {
-      ASFailUpdateValidation(@"Invalid number of items in section %zd. The number of items after the update (%zd) must be equal to the number of items before the update (%zd) plus or minus the number of items inserted or deleted (%zd inserted, %zd deleted).", oldSection, newItemCount, oldItemCount, insertedItemCount, deletedItemCount);
+//      ASFailUpdateValidation(@"Invalid number of items in section %zd. The number of items after the update (%zd) must be equal to the number of items before the update (%zd) plus or minus the number of items inserted or deleted (%zd inserted, %zd deleted).", oldSection, newItemCount, oldItemCount, insertedItemCount, deletedItemCount);
```
Comment (levi) [inline AsyncDisplayKit/Private/_ASHierarchyChangeSet.mm]: you need this?

Comment (Adlai-Holler) [inline AsyncDisplayKit/Private/_ASHierarchyChangeSet.mm]: Lol oops. Kittens has some issues right now. Reverting

**Comment by hannahmbanana** [comment]
double thumbs up!

**Comment by maicki** [comment]
LGTM

**Pull request merged**